<?xml version="1.0"?>

<!--Edited from hybrid_astar/manual.launch-->

<launch>

  <arg name="geotiff_map_file_path" default="$(find wheego)/maps"/>

  <param name="/use_sim_time" value="true"/>

  <node pkg="tf" type="static_transform_publisher"
    name="base_to_laser_broadcaster"
    args="0 0 0 0 0 0 base_footprint velodyne 100"/>

  <include file="$(find wheego)/launch/mapping_edit.launch">
    <arg name="odom_frame" value="base_footprint"/>
  </include>

  <!--<node name="map_mask" pkg="wheego" type="map_mask"/>-->

  <include file="$(find hector_geotiff)/launch/geotiff_mapper.launch">
    <arg name="trajectory_source_frame_name" value="scanmatcher_frame"/>
    <arg name="map_file_path" value="$(arg geotiff_map_file_path)"/>
  </include>

  <!-- Turn on hybrid_astar node -->
  <node name="hybrid_astar" pkg="hybrid_astar" type="hybrid_astar">
    <!--make initalpose change to poseupdate?-->
    <remap from="initialpose" to="poseupdate"/>
    <remap from="map" to="map2"/>
  </node>
  
  <node pkg="tf" type="static_transform_publisher"
    name="odom_to_map2_broadcaster"
    args="0 0 0 0 0 0 odom map2 100"/>
  <node pkg="tf" type="static_transform_publisher"
    name="map2_to_path_broadcaster"
    args="0 0 0 0 0 0 map2 path 100"/>
  <!--
  <node name="tf_broadcaster" pkg="hybrid_astar" type="tf_broadcaster">
    <remap from="map" to="map2"/>
  </node>
  -->

  <node name="rviz" pkg="rviz" type="rviz"
    args="-d $(find wheego)/launch/mapNav.rviz" />
</launch>
